src/__tests__/template-test.js

```diff
@@ -0,0 +1,180 @@
+
+    it('explodes arrays in function definitions', () => {
+      let input = `var foo = [a, b];`;
+      let expected = `var foo = function foo(a, b, c) {};`;
+
+      expect(
+        jscodeshift(input)
+          .find('ArrayExpression')
+          .replaceWith(
+            ({node}) => expression`function foo(${node.elements}, c) {}`
+          )
+          .toSource()
+      )
+      .toEqual(expected);
+
+      expected = `var foo = function(a, b, c) {};`;
+
+      expect(
+        jscodeshift(input)
+          .find('ArrayExpression')
+          .replaceWith(
+            ({node}) => expression`function(${node.elements}, c) {}`
+          )
+          .toSource()
+      )
+      .toEqual(expected);
+
+      expected = `var foo = (a, b) => {};`;
+
+      expect(
+        jscodeshift(input)
+          .find('ArrayExpression')
+          .replaceWith(
+            ({node}) => expression`${node.elements} => {}`
+          )
+          .toSource()
+      )
+      .toEqual(expected);
+
+      expected = `var foo = (a, b, c) => {};`;
+
+      expect(
+        jscodeshift(input)
+          .find('ArrayExpression')
+          .replaceWith(
+            ({node}) => expression`(${node.elements}, c) => {}`
+          )
+          .toSource()
+      )
+      .toEqual(expected);
+    });
+
+    it('explodes arrays in variable declarations', () => {
+      let input = `var foo = [a, b];`;
+      let expected = `var foo, a, b;`;
+      expect(
+        jscodeshift(input)
+          .find('VariableDeclaration')
+          // Need to use a block here because the arrow doesn't seem to be
+          // compiled with a line break after the return statement. Can't repro
+          // outside here though
+          .replaceWith(({node: {declarations: [node]}}) => {
+            return statement`var ${node.id}, ${node.init.elements};`;
+          })
+          .toSource()
+      )
+      .toEqual(expected);
+    });
+
+    it('explodes arrays in array expressions', () => {
+      let input = `var foo = [a, b];`;
+      let expected = `var foo = [a, b, c];`;
+      expect(
+        jscodeshift(input)
+          .find('ArrayExpression')
+          .replaceWith(({node}) => expression`[${node.elements}, c]`)
+          .toSource()
+      )
+      .toEqual(expected);
+    });
+
+    it('explodes arrays in object expressions', () => {
+      let input = `var foo = {a, b};`;
+      let expected = /var foo = \{\s*a,\s*b,\s*c: 42\s*};/;
+      expect(
+        jscodeshift(input)
+          .find('ObjectExpression')
+          .replaceWith(({node}) => expression`{${node.properties}, c: 42}`)
+          .toSource()
+      )
+      .toMatch(expected);
+    });
+
+    it('explodes arrays in call expressions', () => {
+      let input = `var foo = [a, b];`;
+      let expected = `var foo = bar(a, b, c);`;
+
+      expect(
+        jscodeshift(input)
+          .find('ArrayExpression')
+          .replaceWith(
+            ({node}) => expression`bar(${node.elements}, c)`
```

Comment:
If you want to insert a literal array, you would simply build one:

    `bar(${jscodeshift.arrayExpression([...])})`